Formula/influxdb.rb

```diff
@@ -58,6 +58,51 @@ def install
     ldflags = "-s -w -X main.version=#{version}"
     system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"influx", "./cmd/influx"
     system "go", "build", *std_go_args(ldflags: ldflags), "-tags", "assets", "-o", bin/"influxd", "./cmd/influxd"
+
+    # Generate default config file.
+    # TODO: Warn if there's already a config file there?
+    data = var/"lib/influxdb2"
+    data.mkpath
+    (etc/"influxdb2/config.yml").write Utils.safe_popen_read(
+      bin/"influxd", "print-config", "--bolt-path=#{data}/influxdb.bolt", "--engine-path=#{data}/engine"
+    )
+
+    # Create directory for DB stdout+stderr logs.
+    (var/"log/influxdb2").mkpath
+  end
+
+  plist_options manual: "INFLUXD_CONFIG_PATH=#{HOMEBREW_PREFIX}/etc/influxdb2/config.yml influxd"
+
+  def plist
```

Comment:
It's much easier to make this a `service do` block. See https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/php.rb#L325-L331 for example

Reply:
Hmmm I'm having trouble making `service do` work, it doesn't seem to be "sticking" (and `plist_options` are also being ignored).

Would it be ok if I left it as `def plist`, since I can see that it's working?

Reply:
What isn't working for you with a `service` block?

Reply:
I don't have my WIP locally any more, but what I remember is:
* Certain pieces of the block seemed to be ignored, while others were picked up
  * For example, setting a stdout logfile worked, but setting environment variables didn't
* `plist_options` was ignored/overwritten as if I didn't specify it
* No errors/warnings were logged about typos etc., so it was hard for me to guess what I might have done wrong

A few other things are competing for attention at work so I didn't want to spend much time guess-and-checking the right definition. That said, if anyone has suggestions / if anyone could translate the working `plist` into a `service` block that should work, I'd be happy to test it out on Monday.

Reply:
Try this
```ruby
service do
  run opt_bin/"influxd"
  working_dir HOMEBREW_PREFIX
  environment_variables INFLUXD_CONFIG_PATH: etc/"influxdb2/config.yml"
  run_type :immediate
  keep_alive true
  error_log_path var/"log/influxdb2/influxd_output.log"
  log_path var/"log/influxdb2/influxd_output.log"
end
```

Reply:
With that in place instead of `def plist`, I see:
```
==> Caveats
To start influxdb:
  brew services start influxdb
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/influxdb/bin/influxd
```
Instead of the expected:
```
==> Caveats
To have launchd start influxdb now and restart at login:
  brew services start influxdb
Or, if you don't want/need a background service you can just run:
  INFLUXD_CONFIG_PATH=/usr/local/etc/influxdb2/config.yml influxd
```
I didn't modify `plist_options` so I'm not sure why changing to `service do` affects the "manual" example.

Reply:
@SMillerDev, ideas?

Other than the manual command, is the generated `plist` correct?

Reply:
This piece of it looks off, specifically the "data":
```
<key>EnvironmentVariables</key>
<dict>
    <key>INFLUXD_CONFIG_PATH</key>
    <!-- The <data> element below contains a Ruby object which has been serialized with Marshal.dump. -->
    <data>
    BAhlOhNJbnN0YWxsUmVuYW1lZG86DVBhdGhuYW1lBjoKQHBhdGhJIigvdXNyL2xvY2Fs
    L2V0Yy9pbmZsdXhkYjIvY29uZmlnLnltbAY6BkVU
    </data>
</dict>
```
That could explain why the service doesn't use the right data paths when it runs using this plist.

Reply:
The `service` block probably isn't coercing `Pathname`s into `String`s. Try
```
environment_variables INFLUXD_CONFIG_PATH: (etc/"influxdb2/config.yml").to_s
```

Reply:
Yeah, we should probably coerce the environment variables to strings too. (And prefix them to the generated manual command)

Reply:
Is there a way for me to override the generated manual command, so I could use the updated `service do` block but still have the env var in my manual output?

Reply:
I believe passing `plist_options manual: ""` should still work

Reply:
I can see that it doesn't (at least, not with my current code). I have:
```
plist_options manual: "INFLUXD_CONFIG_PATH=#{HOMEBREW_PREFIX}/etc/influxdb2/config.yml influxd"

service do
  run opt_bin/"influxd"
  working_dir HOMEBREW_PREFIX
  environment_variables INFLUXD_CONFIG_PATH: (etc/"influxdb2/config.yml").to_s
  run_type :immediate
  keep_alive true
  error_log_path var/"log/influxdb2/influxd_output.log"
  log_path var/"log/influxdb2/influxd_output.log"
end
```
And I see this as the result:
```
To start influxdb:
  brew services start influxdb
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/influxdb/bin/influxd
```

Reply:
`plist_options manual:` might need to be inside the `service` block?

Reply:
That gives:
```
Error: undefined method `plist_options' for #<Homebrew::Service:0x00007fb339149788>
```

Reply:
Ok then IDK 😄 We can leave the whole `plist` there for now while this needs fixing in `brew`.